lib/daimon_markdown/plugins/toc_pages.rb

```diff
@@ -0,0 +1,101 @@
+module DaimonMarkdown
+  class Plugin
+    class TableOfContentsPages < Base
+      Plugin.register("toc_pages", self)
+
+      include MarkdownHelper
+
+      def call(limit = nil)
+        return unless context.key?(:request)
+
+        request = context[:request]
+
+        current_page = if request.params.key?(:page)
```

Comment:
I want to use full functionalities of `request` in this plugin.
But in fact, this plugin uses some functionalities of `request`.